ui/src/app/utils/runtime-hooks.tsx

```diff
@@ -335,19 +335,33 @@ export const useCustomRuntime = (
           )
         );
 
-        if (mostSevereDiskDiff === AnalysisDiffState.CAN_UPDATE_IN_PLACE) {
+        // update the disk in place, if the runtime can also be updated in place (or needs no change)
+        if (
+          mostSevereDiff <= AnalysisDiffState.CAN_UPDATE_IN_PLACE &&
```

Comment:
bug: we did not account for the situation where the runtime itself needed a significant change which can't be handled by `updateRuntimeAndInitializeLeoRuntime()`

Reply:
less that that is NO_CHANGE, so there is no reason to make an update **due to a disk change**.  Making an update **due to a runtime change** is handled elsewhere in this function.